stdlib/ballerina-database/src/main/java/org/ballerinalang/database/sql/SQLDataIterator.java

```diff
@@ -130,11 +135,19 @@ public BStruct generateNext() {
                     SQLColumnDefinition def = (SQLColumnDefinition) columnDef;
                     columnName = def.getName();
                     int sqlType = def.getSqlType();
+                    BStructType.StructField[] structFields = this.type.getStructFields();
```

Comment:
Shall we change this to `int fieldTypeTag = structFields[index - 1].getFieldType().getTag();`